client/domain/CorrectnessTestObjectFactory.js

```diff
@@ -31,6 +31,17 @@ tie.factory('CorrectnessTestObjectFactory', [
     };
 
     CorrectnessTest.prototype.matchesOutput = function(output) {
+    // If allowOutputs should be arrays and learner's code returns an array
```

Comment:
I'm a little confused by all this. Would the following work just as well?

```
    return this._allowedOutputs.some(function(allowedOutput) {
      return angular.equals(allowedOutput, output);
    });
```

or is there some case where it fails and you need the more complicated version?